--- origsrc/apache-ant-1.8.0/bootstrap.sh	2010-02-01 18:37:47.000000000 -0600
+++ src/apache-ant-1.8.0/bootstrap.sh	2010-02-25 14:33:07.811855100 -0600
@@ -19,7 +19,6 @@
 cygwin=false;
 darwin=false;
 case "`uname`" in
-  CYGWIN*) cygwin=true ;;
   Darwin*) darwin=true
            if [ -z "$JAVA_HOME" ] ; then
              JAVA_HOME=/System/Library/Frameworks/JavaVM.framework/Home
@@ -38,11 +37,11 @@ fi
 # You will need to specify JAVA_HOME if compiling with 1.2 or later.
 
 if [ -n "$JAVA_HOME" ] ; then
-  if [ -f "$JAVA_HOME/lib/tools.jar" ] ; then
+  if [ -e "$JAVA_HOME/lib/tools.jar" ] ; then
     CLASSPATH=$CLASSPATH:$JAVA_HOME/lib/tools.jar
   fi
 
-  if [ -f "$JAVA_HOME/lib/classes.zip" ] ; then
+  if [ -e "$JAVA_HOME/lib/classes.zip" ] ; then
     CLASSPATH=$CLASSPATH:$JAVA_HOME/lib/classes.zip
   fi
 else
@@ -127,7 +126,8 @@ mkdir -p bin
 
 echo ... Compiling Ant Classes
 
-"${JAVAC}" $BOOTJAVAC_OPTS -d ${CLASSDIR} ${TOOLS}/bzip2/*.java ${TOOLS}/tar/*.java ${TOOLS}/zip/*.java \
+"${JAVAC}" -classpath "${CLASSPATH}" $BOOTJAVAC_OPTS -d ${CLASSDIR} \
+    ${TOOLS}/bzip2/*.java ${TOOLS}/tar/*.java ${TOOLS}/zip/*.java \
     ${TOOLS}/ant/util/regexp/RegexpMatcher.java \
     ${TOOLS}/ant/util/regexp/RegexpMatcherFactory.java \
     ${TOOLS}/ant/property/*.java \
--- origsrc/apache-ant-1.8.0/src/script/ant	2010-02-01 18:37:39.000000000 -0600
+++ src/apache-ant-1.8.0/src/script/ant	2010-02-25 14:13:57.799078200 -0600
@@ -82,7 +82,6 @@ cygwin=false;
 darwin=false;
 mingw=false;
 case "`uname`" in
-  CYGWIN*) cygwin=true ;;
   Darwin*) darwin=true
            if [ -z "$JAVA_HOME" ] ; then
              JAVA_HOME=/System/Library/Frameworks/JavaVM.framework/Home
@@ -196,10 +195,10 @@ if $rpm_mode && [ -x /usr/bin/build-clas
 
   # Explicitly add javac path to classpath, assume JAVA_HOME set
   # properly in rpm mode
-  if [ -f "$JAVA_HOME/lib/tools.jar" ] ; then
+  if [ -e "$JAVA_HOME/lib/tools.jar" ] ; then
     LOCALCLASSPATH="$LOCALCLASSPATH:$JAVA_HOME/lib/tools.jar"
   fi
-  if [ -f "$JAVA_HOME/lib/classes.zip" ] ; then
+  if [ -e "$JAVA_HOME/lib/classes.zip" ] ; then
     LOCALCLASSPATH="$LOCALCLASSPATH:$JAVA_HOME/lib/classes.zip"
   fi
 
--- origsrc/apache-ant-1.8.0/src/script/runant.pl	2010-02-01 18:37:47.000000000 -0600
+++ src/apache-ant-1.8.0/src/script/runant.pl	2010-02-25 14:13:57.803078400 -0600
@@ -72,7 +72,7 @@ if ($^O eq "NetWare")
   $onnetware = 1;
 }
 
-my $oncygwin = ($^O eq "cygwin");
+my $oncygwin = 0;
 
 #ISSUE: what java wants to split up classpath varies from platform to platform 
 #and perl is not too hot at hinting which box it is on.
